test/from-old-versions.test.js

```diff
@@ -0,0 +1,55 @@
+const fsPromise = require('fs/promises')
+const test = require('tape')
+const { promisify: p } = require('util')
+
+const OldTestBot = require('scuttle-testbot-1-11-0')
+const NewBot = require('./helpers/test-bot')
+
+const OldBot = (opts) => {
+  let stack = OldTestBot // eslint-disable-line
+    .use(require('ssb-backlinks-2-1-1'))
+    .use(require('ssb-query-2-4-5'))
+    .use(require('ssb-tribes-3-1-3'))
+
+  return stack(opts)
+}
+
+test('can continue from old keyring from ssb-tribes 3.1.3', async t => {
+  const oldAlice = OldBot({ name: 'alice' })
+
+  const oldGroup = await p(oldAlice.tribes.create)({})
+
+  t.equal(typeof oldGroup.groupId, 'string', "got a group id")
+
+  const oldList = await p(oldAlice.tribes.list)()
+  t.deepEqual(oldList, [oldGroup.groupId], 'group got listed')
+
+  // TODO: get group as well
+
+  await p(oldAlice.close)(true)
+
+  await p(setTimeout)(500)
+
+  const newOpts = {
+    name: 'alice',
+    startUnclean: true,
+  }
+
+  //t.throws(() => {
+  //  NewBot(newOpts)
+  //}, /found an old keystore/, "Error when there's an old keystore but we don't use it")
```

Comment:
this seems to work as planned, problem is the bot keeps a lock in flume open, and we can't close this since it returns undefined. manually deleting the lock didn't help.

```
ok 3 Error when there's an old keystore but we don't use it

/home/me/prj/ssb/ssb-tribes/node_modules/levelup/lib/levelup.js:119
      return callback(new OpenError(err))
                      ^
Error [OpenError]: IO error: lock /tmp/alice/flume/keys/LOCK: already held by process
```

if we're lucky this is just a thing from db1, and it won't be a problem once we stop including db1 https://github.com/ssbc/ssb-tribes/pull/105

Reply:
removed db1 in another pr now. unfortunately now getting a very similar error when uncommenting this, just from db2 instead

```
ok 7 Error when there's an old keystore but we don't use it

/home/me/prj/ssb/ssb-tribes/node_modules/levelup/lib/levelup.js:119
      return callback(new OpenError(err))
                      ^
Error [OpenError]: IO error: lock /tmp/alice/db2/indexes/base/LOCK: already held by process
    at /home/me/prj/ssb/ssb-tribes/node_modules/levelup/lib/levelup.js:119:23
    at /home/me/prj/ssb/ssb-tribes/node_modules/abstract-leveldown/abstract-leveldown.js:38:14
    at /home/me/prj/ssb/ssb-tribes/node_modules/deferred-leveldown/deferred-leveldown.js:31:21
    at /home/me/prj/ssb/ssb-tribes/node_modules/abstract-leveldown/abstract-leveldown.js:38:14
    at /home/me/prj/ssb/ssb-tribes/node_modules/abstract-leveldown/abstract-leveldown.js:38:14
Emitted 'error' event on LevelUP instance at:
    at /home/me/prj/ssb/ssb-tribes/node_modules/levelup/lib/levelup.js:60:19
    at /home/me/prj/ssb/ssb-tribes/node_modules/levelup/lib/levelup.js:119:14
    at /home/me/prj/ssb/ssb-tribes/node_modules/abstract-leveldown/abstract-leveldown.js:38:14
    [... lines matching original stack trace ...]
    at /home/me/prj/ssb/ssb-tribes/node_modules/abstract-leveldown/abstract-leveldown.js:38:14 {
  [cause]: undefined
}

Node.js v18.16.0
```

Reply:
fixed by running in a separate process bf6a4f9